alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultProviderDataAccessor.java

```diff
@@ -163,12 +164,21 @@ private void mapUsersToProjectByEmail(Long providerConfigId, String projectHref,
         ProviderProjectEntity project = providerProjectRepository.findFirstByHref(projectHref)
                                             .orElseThrow(() -> new AlertDatabaseConstraintException("A project with the following href did not exist: " + projectHref));
         Long projectId = project.getId();
+        List<ProviderUserProjectRelation> userRelationsToRemove = providerUserProjectRelationRepository.findByProviderProjectId(projectId);
+        List<ProviderUserProjectRelation> userRelationsToAdd = new LinkedList<>();
         for (String emailAddress : emailAddresses) {
-            providerUserRepository.findByEmailAddressAndProviderConfigId(emailAddress, providerConfigId)
-                .stream()
-                .map(ProviderUserEntity::getId)
-                .forEach(userId -> providerUserProjectRelationRepository.save(new ProviderUserProjectRelation(userId, projectId)));
+            List<ProviderUserEntity> providerUserEntities = providerUserRepository.findByEmailAddressAndProviderConfigId(emailAddress, providerConfigId);
```

Comment:
Why does `findByEmailAddressAndProviderConfigId(...)` return a `List`? Shouldn't it be an optional? 
I think what we want is a `findByEmailAddressInAndProviderConfigId(List<String> emailAddresses, Long providerConfigId);

Reply:
The same email address could be a user in multiple configs so it returns a list per the API.  I will add the in keyword and test it out.

Reply:
Right, but we are filtering the config, so that should be a unique combination, no?

Reply:
You can also have multiple blackduck users with the same email address.  We do that for testing. The username is unique in Black Duck but not the email address when it is a local user.  That is why we need a list.

Reply:
Gotcha. Regardless, we should make this a single query.